### ISSUE TITLE
Timeout, no response from Server. We're sorry...

### ISSUE THREAD

**Issue by raimund8323** [codelovers/asana-time-track#41]
![time-track-timeout](https://f.cloud.github.com/assets/6812165/2291612/dc606982-a042-11e3-84b6-d1aefc021587.png)

Please help me with this issue i have tried everything, please advise

**Comment by leonpidgeon** [comment]
Same issue for this with me over the last week or so.

**Comment by michielcazemier** [comment]
We have the same problem since a few weeks. Does anyone has a solution for this?

**Comment by zoephilipps** [comment]
I'm having the same experience.

**Comment by michielcazemier** [comment]
I recently updated my server (MySQL, PHP, Apache), maybe something to do with this?

**Comment by zoephilipps** [comment]
Perhaps. Is your installation on a subdomain?

**Comment by michielcazemier** [comment]
Yes it is, but I also tried to do it via the "normal" link, like domain.com/time

**Comment by Lloyd66** [comment]
Hi!
I encountered the same isssue just after installing it. :(
I executed the request to the API manually and it takes 300 seconds to respond whereas if I make a request directly to Asana it takes less than 1 second to respond.

Any idea where is the problem?
Thank you

**Comment by fourleafed** [comment]
+1
Running from http://localhost/asana-tracker/

**Comment by AntonyXSI** [comment]
I had the same issue and found the only thing that helped was to get a new asana account or delete old archive tasks.

**Comment by pMalvasio** [comment]
The error happens because the API try to get all the tasks, and for huge workspaces consume too much time.
The solution is replace de line 60 in the file AsanaApi.php ( return $this->apiRequest($this->workspaceUri.'/'.$workspaceId.'/tasks?assignee=me'); ) for return $this->apiRequest($this->workspaceUri.'/'.$workspaceId.'/tasks?assignee=me&completed_since=now');

Now only get tasks with due date but no the closed tasks.
If anyone want to do a pull request, go ahead. Hope that solves your problems :)

**Comment by simmo-codes** [comment]
Same issue for me. On large workspaces it fails and can't retrieve tasks. I even increased the timout length in scripts.js by an order of magnitude...still falls over on my large agency workspace.

**Comment by raimund8323** [comment]
Parse error: syntax error, unexpected 'return' (T_RETURN) in /home/hsbnet/public_html/timetrack/AsanaApi.php on line 60.... Please help what is the correct code
